exp/api/v1beta1/azuremachinepool_webhook.go

```diff
@@ -249,22 +250,27 @@ func (amp *AzureMachinePool) ValidateOrchestrationMode(c client.Client) func() e
 	return func() error {
 		// Only Flexible orchestration mode requires validation.
 		if amp.Spec.OrchestrationMode == infrav1.OrchestrationModeType(compute.OrchestrationModeFlexible) {
-			// Find the owner MachinePool
-			ownerMachinePool := &expv1.MachinePool{}
-			key := client.ObjectKey{
-				Namespace: amp.Namespace,
-				Name:      amp.Name,
+			// Find the parent MachinePool
+			var parent *expv1.MachinePool
+			maxAttempts := 5
```

Comment:
would it make sense to implement the retry inside the `FindParentMachinePool` helper function so that @willie-yao doesn't have to repeat this retry logic in #2965 ? Or if we don't want to retry in utils, add a wrapper helper func in webhooks that does the fetch + retry

Reply:
I don't know whether retry is appropriate for that PR, but I can move the retry logic into a wrapper function so its reusable.

Reply:
I think we can implement this inside of the helper function, with an input for how many retries. For my own understanding: why is a retry necessary in this case? Is it because the parent machinepool may not be created at this point?

Reply:
I implemented a [wrapper](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2965/commits/28eaa3755e32b6aa226a7e1a9d5baa99d0c399b3#diff-88cb69b80a9047f0e0cd188951df71c621d843620b72bd52628899b14644209eR116) since there are two places where the error can occur.

Reply:
> why is a retry necessary in this case?

My best guess is that it's what @sbueringer mentioned above: we're using a caching `client.Client` that needs a beat to find the `MachinePool` that was just created (just before the `AzureMachinePool` that triggered this webhook).

Either that, or the actual creation is somehow asynchronous and we're waiting for a controller to finish creating the `MachinePool` object. That seems less likely, because you can see the log line "machinepool.x.y.z yadda yadda created" printed out before the `AzureMachinePool` is processed and this webhook is called.